botrest/src/main/java/com/blackbook/botrest/controller/BooksController.java

```diff
@@ -26,24 +24,14 @@
     @Autowired
     private BooksRepository booksRepository;
 
-    @RequestMapping(method = RequestMethod.GET, value = "/more-expensive")
-    public List<Book> getBooksMoreExpensiveThan(@RequestParam double price) {
-        return booksRepository.findByPriceGreaterThan(price);
-    }
-
-    @RequestMapping(method = RequestMethod.GET, value = "/cheaper")
-    public List<Book> getBooksCheaperThan(@RequestParam double price) {
-        return booksRepository.findByPriceLessThan(price);
-    }
-
     @RequestMapping(method = RequestMethod.GET)
     public Page<Book> getBooks(@RequestParam(defaultValue = "") String query,
                                @RequestParam(required = false) String priceFrom,
                                @RequestParam(required = false) String priceTo,
                                Pageable pageable) {
         log.info("TRANSACTION: GET /api/books");
 
-        if (priceFrom != null && priceTo != null) {
+        if (priceFrom != null && priceTo != null && !priceFrom.isEmpty() && !priceTo.isEmpty()) {
```

Comment:
You should extract a method which would do this evaluation. The name of this method would tell a lot about what this is doing and it would be more readable thanks to that.